lib/pf/Authentication/Source/EmailSource.pm

```diff
@@ -104,20 +107,85 @@ sub mandatoryFields {
 sub authenticate {
     my ( $self, $username, $password ) = @_;
     my $logger = pf::log::get_logger;
+    if (!$self->isEmailAllowed($username)) {
+        $logger->warn("Failed to authenticate using EmailSource ($self->{id}) with PID '$username'");
```

Comment:
IMO, we should give a more precise reason in log, like you did before.
Is it possible to include content of "EMAIL_UNAUTHORIZED" ?

Reply:
Just to say that the authentication failed due to "banned" domain